### DISCUSSION TITLE
Make renovate to respect version range from package.json

### DISCUSSION THREAD

**Discussion by karlismelderis** [renovatebot/renovate#10970] [Request Help]
We decided to keep version range in package.json because that's something that people are used to and introducing another location (renovate.json) felt unnecessary.
Also it's easier to check in git history of single file and follow through why some package was updated/upgraded or locked to fixed version.

Hence we opted for following renovate config:
```ts
"rangeStrategy": "bump"
```

But renovate is still updating packages that we have locked to specific version and don't want to update.

Are we missing some config to finish our setup?

**Comment by rarkins** [comment]
rangeStrategy=bump does not mean that you get no updates, so you'll need to provide more information if you need an answer in any more detail.

**Reply by karlismelderis** [reply]
right, I'm sorry for not giving example first place.

So for example such dependency setup:
```json
  "devDependencies": {
    "packageA": "7.14.8",
    "packageB": "^7.14.5"
}
```

we want renovate to keep on updating `packageB` every time there is new 7.x version and respect pinned version of `packageA` and never try to update it.

and avoid doing any extra version pinning/control outside of package.json

**Reply by rarkins** [reply]
Thanks, now I understand. Currently there is no automatic way to do this - you'd need to add each of these dependencies to a package rule or `ignoreDeps`. If you'd like to see it added, please create a feature request with an example repo like the above which can be tested against. We'll need to figure out the most elegant way to configure this though, e.g. something like `updatePinnedDependencies` which defaults to `true` and setting it to `false` would mean any constraint which is a "single version" would be skipped for updates.

**Reply by karlismelderis** [reply]
Unfortunately repo in question is internal and cannot be shared but I'll open feature request anyways.

**Reply by karlismelderis** [reply]
here it is: https://github.com/renovatebot/renovate/issues/10980

**Reply by rarkins** [reply]
You do not need to share the exact repo, you just need to share an example repo with the same approach
